src/library-authoring/import-course/ImportDetailsPage.tsx

```diff
@@ -0,0 +1,462 @@
+import { HelpSidebar } from './HelpSidebar';
+import { useLibraryContext } from '../common/context/LibraryContext';
+import { useMigrationBlocksInfo } from '../data/apiHooks';
+
+export interface MigrationSummary {
+  totalBlocks: number;
+  sections: number;
+  subsections: number;
+  units: number;
+  components: number;
+  unsupported: number;
+}
+
+export const ImportDetailsPage = () => {
+  const intl = useIntl();
+  const navigate = useNavigate();
+  const { libraryId, libraryData, readOnly } = useLibraryContext();
+  const [enableRefeshState, setEnableRefreshState] = useState(true);
+  const { courseId, migrationTaskId } = useParams();
+  const { showToast } = useContext(ToastContext);
+  const [disableReimport, setDisableReimport] = useState(false);
+
+  // Using bulk migrate as it allows us to create collection automatically
+  // TODO: Modify single migration API to allow create collection
+  const migrate = useBulkModulestoreMigrate();
+
+  if (libraryId === undefined) {
+    // istanbul ignore next - This shouldn't be possible; it's just here to satisfy the type checker.
+    throw new Error('Error: route is missing libraryId.');
+  }
+  if (migrationTaskId === undefined) {
+    // istanbul ignore next - This shouldn't be possible; it's just here to satisfy the type checker.
+    throw new Error('Error: route is missing migrationId.');
+  }
+
+  const {
+    data: courseDetails,
+    isPending: isPendingCourseDetails,
+  } = useCourseDetails(courseId);
+  const {
+    data: migrationStatusData,
+    isPaused: isPendingMigrationStatusData,
+  } = useModulestoreMigrationStatus(migrationTaskId, enableRefeshState ? 1000 : false);
+  // Get the first migration, because the courses are imported one by one
+  const courseImportDetails = migrationStatusData?.parameters?.[0];
+
+  const {
+    data: migrationBlockInfo,
+    isPending: isPendingMigrationBlockInfo,
+    refetch: refetchMigrationBlockInfo,
+  } = useMigrationBlocksInfo(
+    libraryId,
+    undefined,
+    undefined,
+    migrationTaskId,
+    migrationStatusData?.state !== 'Failed',
+  );
+
+  const isPending = isPendingCourseDetails || isPendingMigrationStatusData || isPendingMigrationBlockInfo;
+
+  // Build migration summary using the migration blocks info
+  const {
+    migrationSummary,
+    unsupportedBlockIds,
+  } = useMemo(() => {
+    const counts: MigrationSummary = {
+      totalBlocks: 0,
+      sections: 0,
+      subsections: 0,
+      units: 0,
+      components: 0,
+      unsupported: 0,
+    };
+    const resultUnsupportedIds: string[] = [];
+
+    if (!migrationBlockInfo) {
+      return {
+        migrationSummary: counts,
+        unsupportedBlockIds: resultUnsupportedIds,
+      };
+    }
+
+    for (const block of migrationBlockInfo) {
+      if (!block.targetKey) {
+        // The migrations of this block is failed
+        counts.unsupported += 1;
+        resultUnsupportedIds.push(`"${block.sourceKey}"`);
```

Comment:
Why is each ID `"quoted"` ? If that's required, please add a `/** ... */` JSDoc comment on the line above where `resultUnsupportedIds` is defined, to explain why.

Edit: I see now - it's for the search query. I think it's better to handle it below though.

```
      `usage_key IN [${unsupportedBlockIds.map(k => `"${k}"`).join(',')}]`,

```